pyxem/utils/io_utils.py

```diff
@@ -181,23 +229,40 @@ def load_mib(mib_filename, reshape=True):
     data_pxm.metadata.Signal.exposure_time = data_dict['exposure time']
     data_pxm.metadata.Signal.frames_number_skipped = data_dict['number of frames_to_skip']
     data_pxm.metadata.Signal.flyback_times = data_dict['flyback_times']
-
-    if data_pxm.metadata.Signal.signal_type is 'TEM' and data_pxm.metadata.Signal.exposure_time is not None:
-        print('This mib file appears to be TEM data. The stack is returned with no reshaping.')
-        return data_pxm
-    try:
-        if reshape:
-            # If the exposure time info not appearing in the header bits use reshape_4DSTEM_SumFrames
-            # to reshape otherwise use reshape_4DSTEM_FlyBack function
-            if data_pxm.metadata.Signal.exposure_time is None:
-                data_pxm = reshape_4DSTEM_SumFrames(data_pxm)
-            else:
-                data_pxm = reshape_4DSTEM_FlyBack(data_pxm)
-    except TypeError:
-        raise ValueError('Reshaping did not work. Get the stack by passing reshape=False')
-    except ValueError:
-        raise ValueError('Reshaping did not work. Get the stack by passing reshape=False')
-
+    if reshape:
+    # only attempt reshaping if it is not already reshaped!
+        if len(data_pxm.data.shape) == 3:
+            try:
+                if data_pxm.metadata.Signal.signal_type == 'TEM':
+                    print('This mib file appears to be TEM data. The stack is returned with no reshaping.')
+                    return data_pxm
+                # to catch single frames:
+                if data_pxm.axes_manager[0].size == 1:
+                    print('This mib file is a single frame.')
+                    return data_pxm
+                # If the exposure time info not appearing in the header bits use reshape_4DSTEM_SumFrames
+                # to reshape otherwise use reshape_4DSTEM_FlyBack function
+                if data_pxm.metadata.Signal.signal_type == 'STEM' and data_pxm.metadata.Signal.exposure_time is None:
+                    print('reshaping using sum frames intensity')
+                    (data_pxm, skip_ind) = reshape_4DSTEM_SumFrames(data_pxm)
+                    data_pxm.metadata.Signal.signal_type = 'STEM'
+                    data_pxm.metadata.Signal.frames_number_skipped = skip_ind
+                else:
+                    print('reshaping using flyback pixel')
+                    data_pxm = reshape_4DSTEM_FlyBack(data_pxm)
+            except TypeError:
+                print(
+                    'Warning: Reshaping did not work or TEM data with no exposure info. Returning the stack with no reshaping!')
+                return data_pxm
+            except ValueError:
+                print(
+                    'Warning: Reshaping did not work or TEM data with no exposure info. Returning the stack with no reshaping!')
+                return data_pxm
+    if flip:
+        data_pxm.data = np.flip(data_pxm.data, axis=2)
+        data_pxm.metadata.Signal.flip = True
```

Comment:
I think this might load the data in to memory so doesn't this stop the thing being lazy?

Reply:
Does not seem like it. It still returns a lazy object.

Reply:
It should work providing the correct version of dask and numpy are installed - see
https://numpy.org/neps/nep-0018-array-function-protocol.html#implementation for more information.
Note that dask has a flip function too; I don't know what it is better to use.